spec/models/exchange_rate_currency_rate_spec.rb

```diff
@@ -345,6 +345,69 @@
     it { expect(dataset.pluck(:rate_type)).to eq(%w[monthly]) }
   end
 
+  describe '.by_currency' do
+    subject(:dataset) { described_class.by_currency(currency_code) }
+
+    before do
+      create(:exchange_rate_currency_rate, :with_usa)
+      create(:exchange_rate_currency_rate, :with_eur)
+    end
+
+    context 'when currency code is lower case' do
+      let(:currency_code) { 'usd' }
+
+      it { expect(dataset.pluck(:currency_code)).to eq(%w[USD]) }
+    end
+
+    context 'when currency code is upper case' do
+      let(:currency_code) { 'USD' }
+
+      it { expect(dataset.pluck(:currency_code)).to eq(%w[USD]) }
+    end
```

Comment:
Maybe this is redundant, up to you.

Reply:
Possibly but it was put in place as the code provides giving lower upper case options?